### PR TITLE
Refactor alreadyBookedDays with mongoDB

### PR DESCRIPTION
**Pull request by taniafecheta**

### CODE REVIEW

**Review by climandrei94 — DISMISSED** [review]

File: src/actions/retriveAlreadyBookedDaysActions.js [hunk 1]
```diff
@@ -1,77 +1,89 @@
...
-    });
-    return daysBooked;
-};
+const DaysOff = require("../models/daysOffModel")
+const User = require("../models/userModel")
+const findBookedDays = (userId) => {
+    return new Promise((resolve, reject) => {
+        findDaysOff = DaysOff.find({ userId: userId }, "daysOff", (err, daysOff) => {
+            if (err) reject(err)
+            resolve(JSON.parse(JSON.stringify(daysOff)))
```
Comment (andreicioromila) [inline src/actions/retriveAlreadyBookedDaysActions.js]: What type does `daysOff` have when it returns from the db? What is the purpose of JSON transformation?

File: src/actions/retriveAlreadyBookedDaysActions.js [hunk 2]
```diff
@@ -1,77 +1,89 @@
...
+        alreadyBookedDays.push(...i.daysOff)
+    }
+    return alreadyBookedDays
+}
+
+const getAlreadyBookedDays = (userId) => {
+    return findBookedDays(userId)
+        .then(createArrayOffBookedDaysOff)
+        .then(data => { return data })
+        .catch(err => { throw err })
```
Comment (andreicioromila) [inline src/actions/retriveAlreadyBookedDaysActions.js]: Do you need these last two lines?

File: src/actions/retriveAlreadyBookedDaysActions.js [hunk 3]
```diff
@@ -1,77 +1,89 @@
...
-                }
-            });
-            resolve();
+                        }`,
+                },
+            })
+            resolve()
         } catch (err) {
-            reject(err);
+            reject(err)
```
Comment (andreicioromila) [inline src/actions/retriveAlreadyBookedDaysActions.js]: [experiment] What would happen if the code in send response throws an error? Who catches it? What error would that be?

File: src/actions/retriveAlreadyBookedDaysActions.js [hunk 4]
```diff
@@ -1,77 +1,89 @@
...
-        if (bookedDays.length === 0)
+const checkUserExistsInDB = (userId) => {
+    return new Promise((resolve, reject) => {
+        let findUser = User.findById(userId, (err, user) => {
+            if (err)
+                reject({
+                    status: 400,
+                    message: "Bad user id"
+                });
+            if (user) resolve();
```
Comment (andreicioromila) [inline src/actions/retriveAlreadyBookedDaysActions.js]: Remember the discussion regarding interrupting the flow when you want to

**Review by andreicioromila — CHANGES_REQUESTED** [review]

File: src/actions/retriveAlreadyBookedDaysActions.js (hunk 1 above)
Comment (MihaiBojescu) [reply src/actions/retriveAlreadyBookedDaysActions.js]: We didn't need to transform it, so we removed this functionality.

File: src/actions/retriveAlreadyBookedDaysActions.js (hunk 4 above)
Comment (MihaiBojescu) [reply src/actions/retriveAlreadyBookedDaysActions.js]: Fixed it in latest commits.

File: src/actions/retriveAlreadyBookedDaysActions.js (hunk 2 above)
Comment (MihaiBojescu) [reply src/actions/retriveAlreadyBookedDaysActions.js]: Fixed in the latest commits.

File: tests/retriveAlreadyBookedActions.test.js [hunk 5]
```diff
@@ -1,60 +1,89 @@
...
-
-            chai.expect(bookedDays).to.deep.equal(result);
-            done()
+        it("Should return an array of booked days for an user", done => {
+            let findUser = User.findOne({ email: "costel@gmail.com" })
+                .then((user) => {
+                    let userId = user["_id"];
+                    getAlreadyBookedDays(userId)
+                        .then((arrayOffBookedDays) => {
+                            let alreadyBookedDays = [
```
Comment (andreicioromila) [inline tests/retriveAlreadyBookedActions.test.js:57]: can you chain these promises?

e.g.
```javascript
User.findOne({ email: "costel@gmail.com" })
	.then(user => return user["_id"])
	.then(userid => return getAlreadyBookedDays(userId))
	.then((arrayOffBookedDays) => {
		...
	})
```

**Review by andreicioromila — CHANGES_REQUESTED** [review]

**Review by andreicioromila — DISMISSED** [review]

File: tests/retriveAlreadyBookedActions.test.js (hunk 5 above)
Comment (MihaiBojescu) [reply tests/retriveAlreadyBookedActions.test.js:57]: We placed an item in our TODO list for this issue.

**Review by climandrei94 — DISMISSED** [review]

**Review by MihaiBojescu — DISMISSED** [review]

**Review by climandrei94 — APPROVED** [review]

**Review by MihaiBojescu — APPROVED** [review]

**Review by danielsandu1 — APPROVED** [review]

**Pull request merged**